core/appeal/service.go

```diff
@@ -323,6 +326,25 @@ func (s *Service) Create(ctx context.Context, appeals []*domain.Appeal, opts ...
 	return nil
 }
 
+func (s *Service) findActiveGrant(ctx context.Context, a *domain.Appeal) (*domain.Grant, error) {
+	grants, err := s.grantService.List(ctx, domain.ListGrantsFilter{
+		Statuses:    []string{string(domain.GrantStatusActive)},
+		AccountIDs:  []string{a.AccountID},
+		ResourceIDs: []string{a.ResourceID},
+		Roles:       []string{a.Role},
```

Comment:
filtering active grants by `Role` should be avoided since we add `Permissions` into appeal (https://github.com/odpf/guardian/issues/205), can we also update this filter to use `Permissions` instead of `Role`? 

Note: this might resulting an increase in the query time since permissions column type is an array of string

Reply:
I think, we can add another filter option for `permissions` if needed, but we should still have filtering at `role` level. Since account_id, resource_id, role are supposed to be a unique combination. 
I feel that adding permission here may bring out some false positives since some permissions can be common across different roles, and we would need to add checks for that again

Reply:
the thing with role is that it can be updated in the provider config, so it is possible for an existing active grant and an appeal **with the same role id are having different set of permissions**. 

Using only permissions to filter active grant in this case would tells whether the user already have those permissions in the provider (regardless of the selected role id).

Reply:
current code are using this filter, i'm avoiding changing logic other than performance improvement in this PR
we can create separate PR for that, wdyt?
https://github.com/odpf/guardian/blob/2db7c75325dcaf28871dfd8aaa3956fa878272ab/core/appeal/service.go#L943-L946

Reply:
yes this definitely can be detached to separate issue 👍